Rotator/MaxStorageSizeRotator.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Zenstruck\BackupBundle\Rotator;
+
+use \Exception;
+use Zenstruck\BackupBundle\Destination\Backup;
+use Zenstruck\BackupBundle\Utils\Filesize;
+
+final class MaxStorageSizeRotator implements Rotator
+{
+    /**
+     * @var integer
+     */
+    private $maxSize;
+
+    /**
+     * A constructor.
+     *
+     * @param integer|string $maxSize Maximum storage size.
+     */
+    public function __construct($maxSize)
+    {
+        if (is_numeric($maxSize)) {
+            $this->maxSize = $maxSize;
+        } else {
+            if (class_exists('\\ByteUnits\\System')) {
```

Comment:
Ok, there is optional usage of \ByteUnits, but we will fallback to our own implementation, because common use case: 100GB, 10GB, 500MB -> it is really easy to implement.